### ISSUE TITLE
Intermittent unhandled API exception with AWS SSO admin API 

### ISSUE THREAD

**Issue by leelalagudu** [aws-samples/aws-sso-extensions-for-enterprise#44]
We've noticed a particular issue with AWS SSO admin API:

- When we try and update managed policies list attached to a permission set i.e. run multiple async `DetachManagedPolicyFromPermissionSet` and `AttachManagedPolicyToPermissionSet` calls, the API sometimes responds with an unhandled exception. The exception class is undefined which is evidenced by the V3 JS SDK client failing to parse the response and throwing `TypeError: Converting circular structure to JSON` message. The solution currently has an exponential back-off based retry mechanism, however since this exception is not even understood by the SDK client, the retry isn't helping as well. This is therefore causing incorrect behaviour where an update of the managed policies array is not being processed completely. 
- We've already raised an internal ticket with the AWS SSO service team for this issue and are working with them to provide related logs so that the service team has all the details to understand the root cause.
- #38  also indicated that the same kind of unhandled API exceptions are thrown by AWS SSO admin API even during an `Identity store` describe user API call. 
- Until the service team identifies the root cause and provides recommendations on how to handle this, we're blocked on this issue.

**Comment by leelalagudu** [comment]
We had the following update from the service team:

- The `Converting circular structure to JSON` error and #43 are both symptoms of another issue. The V3 AWS JS SDK is unable to handle exceptions being thrown by the service layer, and the parse failure is being shown as the circular error we are seeing
- For this specific error i.e. attach and detach managed policies issue, the service team advised that we serialise this operation instead of running them in parallel. We've made the change on our local dev and validated that the issue no longer exists with this design change.
- So the exception parsing error would be tracked in #43 , while this issue would be fixed by #49
